app/src/main/java/com/twilio/twiliochat/chat/channels/ChannelExtractor.java

```diff
@@ -0,0 +1,53 @@
+package com.twilio.twiliochat.chat.channels;
+
+import com.twilio.chat.CallbackListener;
+import com.twilio.chat.Channel;
+import com.twilio.chat.ChannelDescriptor;
+import com.twilio.chat.ErrorInfo;
+import com.twilio.chat.Paginator;
+import com.twilio.twiliochat.chat.listeners.TaskCompletionListener;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class ChannelExtractor {
+
+  public void extractAndSortFromChannelDescriptor(Paginator<ChannelDescriptor> paginator, final TaskCompletionListener<List<Channel>, String> listener) {
+    extractFromChannelDescriptor(paginator, new TaskCompletionListener<List<Channel>, String>() {
+      @Override
+      public void onSuccess(List<Channel> channels) {
+        Collections.sort(channels, new CustomChannelComparator());
+        listener.onSuccess(channels);
+      }
+
+      @Override
+      public void onError(String s) {
+        listener.onError(s);
+      }
+    });
+  }
+
+  private void extractFromChannelDescriptor(Paginator<ChannelDescriptor> paginator, final TaskCompletionListener<List<Channel>, String> listener) {
```

Comment:
ditto - the rule applied for determine the max length of a line is how the code is shown in the docs. The maximum length is 97 chars.